kornia/geometry/epipolar/projection.py

```diff
@@ -108,6 +108,43 @@ def projection_from_KRt(K: torch.Tensor, R: torch.Tensor, t: torch.Tensor) -> to
     return K @ Rt
 
 
+def KRt_from_projection(P: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    r"""This function decomposes the Projection matrix into Camera-Matrix, Rotation Matrix and Translation vector.
+
+    Args:
+        P (torch.Tensor): the projection matrix with shape :math:`(B, 3, 4)`.
+
+    Returns:
+        Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        - The Camera matrix with shape :math:`(B, 3, 3)`.
+        - The Rotation matrix with shape :math:`(B, 3, 3)`.
+        - The Translation vector with shape :math:`(B, 3)`.
+
+    """
+
+    assert P.shape[-2:] == (3, 4), "P must be of shape [B, 3, 4]"
+
+    submat_3x3 = P[:, 0:3, 0:3]
+    last_column = P[:, 0:3, 3].unsqueeze(-1)
+    t = torch.matmul(-torch.inverse(submat_3x3), last_column).squeeze(-1)
+
+    # Trick to turn QR-decomposition into RQ-decomposition
+    reverse = torch.Tensor([[0, 0, 1], [0, 1, 0], [1, 0, 0]], device=P.device, dtype=P.dtype).unsqueeze(0)
+    submat_3x3 = torch.matmul(reverse, submat_3x3).permute(0, 2, 1)
+    ortho_mat, upper_mat = torch.qr(submat_3x3)
+    ortho_mat = torch.matmul(reverse, ortho_mat.permute(0, 2, 1))
+    upper_mat = torch.matmul(reverse, torch.matmul(upper_mat.permute(0, 2, 1), reverse))
+
+    # Turning the `upper_mat's` diagonal elements to positive.
+    diagonals = torch.diagonal(upper_mat, dim1=-2, dim2=-1)
+    signs = diagonals / diagonals.abs()
```

Comment:
you might need to add an epsilon value to the denominator. Besides, pytorch already provides [`torch.sign`](https://pytorch.org/docs/stable/generated/torch.sign.html?highlight=sign#torch.sign), but not quite sure about its differentiability property

Reply:
Yes, `torch.sign` is differentiable. I have replaced the previous code with `torch.sign()`. 
Also added the `epsilon` to `diagonals` to prevent elements of `signs` being `zero.`